### PR TITLE
Add Lecture 1 Task preliminary implementation

### PR DESCRIPTION
**Pull request by carmenhorendum**
Task 3 is not completed yet. Task 5 partially works.

### CODE REVIEW

File: Lecture1/Task1/packOne/TaskOneMain.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+		Scanner input = new Scanner(System.in);
+		String string = input.nextLine().trim();
+		String newString = string;
+		String search = " ";
+		String result;
+		int i;
+		StringBuffer stringBuffer = new StringBuffer(string);
+		System.out.println();
+		System.out.println("Original: "+ string);
+		System.out.println();
```
Comment (nemec8g) [inline Lecture1/Task1/packOne/TaskOneMain.java]: System.out.println(); - to private static void printEmptyLine() - to better readability

File: Lecture1/Task1/packOne/TaskOneMain.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import java.util.Scanner;
+
+public class TaskOneMain {
+	public static void main(String[] args) {
+		System.out.println("Type in any sentence: ");
+		Scanner input = new Scanner(System.in);
+		String string = input.nextLine().trim();
+		String newString = string;
+		String search = " ";
```
Comment (nemec8g) [inline Lecture1/Task1/packOne/TaskOneMain.java]: to private static final int SPACE

File: Lecture1/Task1/packOne/TaskOneMain.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+		String result;
+		int i;
```
Comment (nemec8g) [inline Lecture1/Task1/packOne/TaskOneMain.java]: declare var  (result, i) only in place when you need it

File: Lecture1/Task1/packOne/TaskOneMain.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+		StringBuffer stringBuffer = new StringBuffer(string);
+		System.out.println();
+		System.out.println("Original: "+ string);
+		System.out.println();
+		System.out.println("Reversed: "+stringBuffer.reverse());
+		System.out.println();
+		System.out.println("One line - one word: ");
+		do {
```
Comment (nemec8g) [inline Lecture1/Task1/packOne/TaskOneMain.java]: use String.split  method

File: Lecture1/Task1/packOne/TaskOneMain.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+			i = newString.indexOf(search);
+			if (i != -1){
+				result = newString.substring(0,i);
+				System.out.println(result);
+				newString = newString.substring(i + search.length());
+			}
+		} while (i != -1);
+		System.out.println(newString);
+		System.out.println();
+		System.out.println("Spaces to Asterisks: "+ string.replace(' ', '*'));
```
Comment (nemec8g) [inline Lecture1/Task1/packOne/TaskOneMain.java]: i suggest to create few private methods for each subtask

File: Lecture1/Task2/packTwo/TaskTwoMain.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+import java.text.DateFormatSymbols;
+import java.util.Scanner;
+
+public class TaskTwoMain {
+	public static void main(String[] args) {
+		String dayNames[] = new DateFormatSymbols().getWeekdays();
+		System.out.println("Enter weekday number (1-7):");
+		Scanner input = new Scanner(System.in);
+		int dayNumber = Integer.parseInt(input.next());
```
Comment (nemec8g) [inline Lecture1/Task2/packTwo/TaskTwoMain.java]: input.nextInt()

File: Lecture1/Task2/packTwo/TaskTwoMain.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+		System.out.println(String.format("Your day number is %d - %s.", dayNumber, dayNames[dayNumber]));
```
Comment (nemec8g) [inline Lecture1/Task2/packTwo/TaskTwoMain.java:19]: IndexOutOfBoundsException when user entered value more than 7

File: Lecture1/Task3/packThree/TaskThreeMain.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+		int middleStarCount;
+//		int lines;
+//		int length;
+//		int i;
+		String space = " ";
+		String asterisk = "*";
+		System.out.println("Enter size:");
+		Scanner input = new Scanner(System.in);
+		middleStarCount = Integer.parseInt(input.next());
+//		lines = size+1;
```
Comment (nemec8g) [inline Lecture1/Task3/packThree/TaskThreeMain.java]: ?????

File: Lecture1/Task4/packFour/Runner.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
+package packFour;
+
+import java.util.List;
+
+public class Runner {
+	private String name;
```
Comment (nemec8g) [inline Lecture1/Task4/packFour/Runner.java:6]: final

File: Lecture1/Task4/packFour/Runner.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+	private int time;
```
Comment (nemec8g) [inline Lecture1/Task4/packFour/Runner.java:7]: final

File: Lecture1/Task4/packFour/TaskFourMain.java [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+import java.util.List;
+import java.util.Scanner;
+
+public class TaskFourMain {
+	public static void main(String[] args) {
+		String[] names = { "Elena", "Thomas", "Hamilton", "Suzie", "Phil", "Matt", "Alex", "Emma", "John", "James", "Jane", "Emily", "Daniel", "Neda", "Aaron", "Kate" };
+		int[] timeValues = { 341, 273, 278, 329, 445, 402, 388, 275, 243, 334, 412, 393, 299, 343, 317, 265 };
+		List<Runner> runners = new ArrayList<>();
+		for (int i = 0; i < timeValues.length; i++) {
+			runners.add(new Runner(names[i],timeValues[i]));
```
Comment (nemec8g) [inline Lecture1/Task4/packFour/TaskFourMain.java]: what will happened if timeValues.size != names.size ?

File: Lecture1/Task4/packFour/TaskFourMain.java [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+		}
+		runners.sort(Comparator.comparingInt(Runner::getTime));
+		System.out.println();
+		for (int i = 0; i < runners.size(); i++) {
+			runners.get(i).setPlace(i+1);
```
Comment (nemec8g) [inline Lecture1/Task4/packFour/TaskFourMain.java]: runners.get(i) - to local var

File: Lecture1/Task4/packFour/TaskFourMain.java [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+			System.out.println(String.format("Name: %s, Time: %d, Place: %d",runners.get(i).getName(),runners.get(i).getTime(),runners.get(i).getPlace()));
+		}
+		System.out.println();
+		System.out.println("Enter runner place to get info:");
```
Comment (nemec8g) [inline Lecture1/Task4/packFour/TaskFourMain.java]: to separate method private static String  getPlace()

File: Lecture1/Task4/packFour/TaskFourMain.java [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+		Scanner input = new Scanner(System.in);
+		int anyPlace = input.nextInt();
+		int index = anyPlace-1;
+		System.out.println(String.format("Place %d",anyPlace));
+		System.out.println(String.format("Name: %s, Time: %d",runners.get(index).getName(),runners.get(index).getTime()));
```
Comment (nemec8g) [inline Lecture1/Task4/packFour/TaskFourMain.java]: what about winner?

File: Lecture1/Task5/packFive/TaskFiveMain.java [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+		Set picked = new HashSet();
+		while (!mask.equals(answer)) {
+			System.out.println("Make your guess. You can enter a letter or a word:");
+			Scanner input = new Scanner(System.in);
+			String guess = input.nextLine().toLowerCase().trim();
+			int caseId = 0;
+			if (picked.contains(guess)) {
+				caseId = 3;
+			} else {
+//				if (guess.length()==1) {
```
Comment (nemec8g) [inline Lecture1/Task5/packFive/TaskFiveMain.java]: ??

File: Lecture1/Task5/packFive/TaskFiveMain.java [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+						}
+					}
+//				} else {
+//					System.out.println(guess.length());
+//				}
+			}
+			System.out.println(caseId);
+			switch (caseId) {
+				case 1:
+					picked.add(guess);
```
Comment (nemec8g) [inline Lecture1/Task5/packFive/TaskFiveMain.java]: logic with cases - looks not clear for me - better create few private methods and call them directly when it needed

**Review by nemec8g — CHANGES_REQUESTED** [review]

File: Lecture1/Task4/packFour/TaskFourMain.java (hunk 11 above)
Comment (carmenhorendum) [reply Lecture1/Task4/packFour/TaskFourMain.java]: When names.size > timeValues.size then name without time pair will not be added to runners list.
When names.size < timeValues.size then IndexOutOfBoundsException.

File: Lecture1/Task4/packFour/TaskFourMain.java [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+			Runner valueByIndex = runners.get(i);
+			valueByIndex.setPlace(i+1);
+			System.out.println(String.format("Name: %s, Time: %d, Place: %d", 
+						valueByIndex.getName(), valueByIndex.getTime(), valueByIndex.getPlace()));
+		}
+		System.out.println(String.format("\nThe winner is: %s!", runners.get(0).getName()));
+		getInfoByPlace(runners);
+	}
+	
+	private static void getInfoByPlace(List<Runner> runners) {
```
Comment (nemec8g) [inline Lecture1/Task4/packFour/TaskFourMain.java]: method with name `get....` should have some return  value

File: Lecture1/Task4/packFour/TaskFourMain.java [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+		System.out.println("\nEnter place to get runner info:");
+		Scanner input = new Scanner(System.in);
+		int anyPlace = input.nextInt();
+		int index = anyPlace - 1;
+		System.out.println(String.format("Place %d", anyPlace));
+		System.out.println(String.format("Name: %s, Time: %d",
```
Comment (nemec8g) [inline Lecture1/Task4/packFour/TaskFourMain.java]: you will get an exception if anyPlace > runners.size()

File: Lecture1/Task5/packFive/TaskFiveMain.java [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+		String mask = "****** *****";
+		char[] answerArray = answer.toUpperCase().toCharArray();
+		StringBuffer guessed = new StringBuffer(mask);
+		Set picked = new HashSet();
+		while (!mask.equals(answer)) {
+			System.out.println();
+			System.out.println(String.format("\nMake your guess: %s. You can enter a letter or a word.", mask));
+			Scanner input = new Scanner(System.in);
+			String guess = input.nextLine().toUpperCase();
+			int caseId = 0;
```
Comment (nemec8g) [inline Lecture1/Task5/packFive/TaskFiveMain.java]: no needed local var

File: Lecture1/Task5/packFive/TaskFiveMain.java [hunk 20]
```diff
@@ -0,0 +1,71 @@
...
+import java.util.Scanner;
+import java.util.Set;
+
+public class TaskFiveMain {
+	public static void main(String[] args) {
+		String answer = "ДЗЯНІС ГОЛІН";
+		String mask = "****** *****";
+		char[] answerArray = answer.toUpperCase().toCharArray();
+		StringBuffer guessed = new StringBuffer(mask);
+		Set picked = new HashSet();
```
Comment (nemec8g) [inline Lecture1/Task5/packFive/TaskFiveMain.java]: do not use row type 
Set'<'String'>' picked = new HashSet();

File: Lecture1/Task5/packFive/TaskFiveMain.java [hunk 21]
```diff
@@ -0,0 +1,71 @@
...
+	}
+
+	private static void getRepeated(Set picked, String guess) {
+		picked.add(guess);
+		System.out.println(String.format("You've already picked %s! Try something different.", guess));
+	}
+
+	private static void getWin(String answer) {
+		System.out.println(String.format("\nYou win! Correct answer is: %s", answer));
+	}
```
Comment (nemec8g) [inline Lecture1/Task5/packFive/TaskFiveMain.java:69]: please answer on all comments in PR like 'done' 'fix' 'wont fix' etc

File: Lecture1/Task5/packFive/TaskFiveMain.java [hunk 22]
```diff
@@ -0,0 +1,71 @@
...
+					} else {
+						getBadGuess(picked, guess, guessed);
+					}
+				}
+			}
+		}
+		getWin(answer);
+	}
+
+	private static void getBadGuess(Set picked, String guess, StringBuffer guessed) {
```
Comment (nemec8g) [inline Lecture1/Task5/packFive/TaskFiveMain.java]: not `get` . - `update` or something else

File: Lecture1/Task1/packOne/TaskOneMain.java (hunk 1 above)
Comment (carmenhorendum) [reply Lecture1/Task1/packOne/TaskOneMain.java]: done

File: Lecture1/Task1/packOne/TaskOneMain.java (hunk 2 above)
Comment (carmenhorendum) [reply Lecture1/Task1/packOne/TaskOneMain.java]: won't fix (obsolete)

File: Lecture1/Task1/packOne/TaskOneMain.java (hunk 3 above)
Comment (carmenhorendum) [reply Lecture1/Task1/packOne/TaskOneMain.java]: done

File: Lecture1/Task1/packOne/TaskOneMain.java (hunk 4 above)
Comment (carmenhorendum) [reply Lecture1/Task1/packOne/TaskOneMain.java]: done

File: Lecture1/Task1/packOne/TaskOneMain.java (hunk 5 above)
Comment (carmenhorendum) [reply Lecture1/Task1/packOne/TaskOneMain.java]: done

File: Lecture1/Task2/packTwo/TaskTwoMain.java (hunk 6 above)
Comment (carmenhorendum) [reply Lecture1/Task2/packTwo/TaskTwoMain.java]: done

File: Lecture1/Task2/packTwo/TaskTwoMain.java (hunk 7 above)
Comment (carmenhorendum) [reply Lecture1/Task2/packTwo/TaskTwoMain.java:19]: done

File: Lecture1/Task3/packThree/TaskThreeMain.java (hunk 8 above)
Comment (carmenhorendum) [reply Lecture1/Task3/packThree/TaskThreeMain.java]: done

File: Lecture1/Task4/packFour/Runner.java (hunk 9 above)
Comment (carmenhorendum) [reply Lecture1/Task4/packFour/Runner.java:6]: IDE throws and exception that variable name can't be updated if it is declared as final

File: Lecture1/Task4/packFour/Runner.java (hunk 10 above)
Comment (carmenhorendum) [reply Lecture1/Task4/packFour/Runner.java:7]: IDE throws and exception that variable name can't be updated if it is declared as final

File: Lecture1/Task4/packFour/TaskFourMain.java (hunk 12 above)
Comment (carmenhorendum) [reply Lecture1/Task4/packFour/TaskFourMain.java]: done

File: Lecture1/Task4/packFour/TaskFourMain.java (hunk 14 above)
Comment (carmenhorendum) [reply Lecture1/Task4/packFour/TaskFourMain.java]: done

File: Lecture1/Task4/packFour/TaskFourMain.java (hunk 13 above)
Comment (carmenhorendum) [reply Lecture1/Task4/packFour/TaskFourMain.java]: done

File: Lecture1/Task5/packFive/TaskFiveMain.java (hunk 16 above)
Comment (carmenhorendum) [reply Lecture1/Task5/packFive/TaskFiveMain.java]: done

File: Lecture1/Task5/packFive/TaskFiveMain.java (hunk 22 above)
Comment (carmenhorendum) [reply Lecture1/Task5/packFive/TaskFiveMain.java]: fixed

File: Lecture1/Task5/packFive/TaskFiveMain.java (hunk 19 above)
Comment (carmenhorendum) [reply Lecture1/Task5/packFive/TaskFiveMain.java]: fixed

File: Lecture1/Task5/packFive/TaskFiveMain.java (hunk 20 above)
Comment (carmenhorendum) [reply Lecture1/Task5/packFive/TaskFiveMain.java]: fixed

File: Lecture1/Task4/packFour/TaskFourMain.java (hunk 17 above)
Comment (carmenhorendum) [reply Lecture1/Task4/packFour/TaskFourMain.java]: fixed

File: Lecture1/Task4/packFour/TaskFourMain.java (hunk 18 above)
Comment (carmenhorendum) [reply Lecture1/Task4/packFour/TaskFourMain.java]: fixed

File: Lecture1/Task5/packFive/TaskFiveMain.java (hunk 21 above)
Comment (carmenhorendum) [reply Lecture1/Task5/packFive/TaskFiveMain.java:69]: done

**Pull request merged**